packages/es-components/src/components/controls/switch/Switch.js

```diff
@@ -0,0 +1,259 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { noop } from 'lodash';
+
+const Directions = {
+  'top': 'column-reverse',
+  'bottom': 'column',
+  'right': 'row',
+  'left': 'row-reverse'
+}
+
+const Padding = {
+  'top': '0 0 0.5rem',
+  'bottom': '2.25rem 0 0',
+  'right': '0 0 0.5rem',
+  'left': '0 1rem 0 0'
+}
+
+const primary = "#0073b6";
+const primaryLight = "#ededed";
+const success = "#298544";
+const successLight = "#69aa7c";
+const warning = "#c25400";
+const warningLight = "#d4874d";
+const danger = "#cc0000";
+const dangerLight = "#db4d4d";
+
+function Type(type, light) {
+  switch (type) {
+    case "primary":
+      return light ? primaryLight : primary;
+    case "success":
+      return light ? successLight : success;
+    case "warning":
+      return light ? warningLight : warning;
+    case "danger":
+      return light ? dangerLight : danger;
+    default:
+      return light ? primaryLight : primary;
+  }
+}
+
+const SwitchBase = styled.div`
+  margin: 0 auto 10px;
+  display: flex;
+
+  label {
+    cursor: pointer;
```

Comment:
Could any of these properties be moved directly to their respective styled-components? i.e. move these to `SwitchLabel` instead of the `label` here, or `SwitchInput` for the styles nested under `input[type="checkbox"]`?

Reply:
Yeah, wasn't thinking. I was working off a SASS document. All clean now.